ddtrace/tracer/tracer.go

```diff
@@ -400,8 +423,46 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 		// if not already sampled or a brand new trace, sample it
 		t.sample(span)
 	}
+	if t.config.profilerHotspots || t.config.profilerEndpoints {
+		t.applyPPROFLabels(pprofCtx, span)
+	}
 	log.Debug("Started Span: %v, Operation: %s, Resource: %s, Tags: %v, %v", span, span.Name, span.Resource, span.Meta, span.Metrics)
-	return span
+	return span, ContextWithSpan(ctx, span)
+}
+
+// applyPPROFLabels applies pprof labels for the profiler's code hotspots
+// and endpoint filtering feature.
+func (t *tracer) applyPPROFLabels(ctx gocontext.Context, span *span) {
```

Comment:
I would normally do it the other way around, `append`-style. Putting the `span` first, and `ctx` second. I know `ctx` is usually first but that's in a different usage context. I don't feel strongly on this so I'll let you decide. But I feel like `span` is the main actor here, and `ctx` is secondary.

Reply:
Hmm. I see your point, but I think I'll leave it as-is if you're okay with it. Arguably the main operation is applying the labels to the current goroutine. Mutating the span is just a necessary side-quest. So overall I still prefer the ctx first convention here.